homework/list.js

```diff
@@ -1,79 +1,83 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
-
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
-
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
-
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
+	var ListManager = function();
+	 this._listNoteContent = [];
+	 this._wrapper = document.querySelector('#note-list-wrapper');
   }
 
-  function updateList(list) {
-    _listNoteContent = list;
-  }
+  ListManager.prototype = {
+  start(){
+  this.fetchList()
+  .then((function)(data){
+  this.updateList(data);
+  this.drawList();
+  this.preloadFirstNote();
+  }).bind(this))
+  .catch((function(){
+  
+  }).bind(this) );
+  
+  window.addEventListener('click',(function(event){
+  this.onNote(event);
+  }).bind(this))
+  },
 
-  function drawList() {
-    var list = _listNoteContent;
-    var ul = document.createElement('ul');
-    ul.id = 'note-title-list';
-    var buff = document.createDocumentFragment();
-    list.forEach(function(note, i) {
-      var li = document.createElement('li');
-      li.dataset.noteId = i;
-      li.classList.add('note-title');
-      li.textContent = note.title;
-      // Note: buff is captured, so we now have a
-      // little closure naturally.
-      buff.appendChild(li);
-    });
-    ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
+  onNoteOpen(event){
```

Comment:
It's a convention to prefix method with on for the corresponding function, so it should be renamed as 'onClick'